icu.macro.js

```diff
@@ -85,39 +68,54 @@ function pluralAsJSX(parentPath, { attributes }, babel) {
   // plural or selectordinal
   const nodeName = parentPath.node.name.name.toLocaleLowerCase();
 
-  let componentStartIndex = 0;
+  // will need to merge count attribute with existing values attribute in some cases
+  const existingValuesAttribute = findAttribute('values', attributes);
+  const existingValues = existingValuesAttribute
+    ? existingValuesAttribute.node.value.expression.properties
+    : [];
 
+  let componentStartIndex = 0;
   const extracted = attributes.reduce(
     (mem, attr) => {
       if (attr.node.name.name === 'i18nKey') {
         // copy the i18nKey
         mem.attributesToCopy.push(attr.node);
       } else if (attr.node.name.name === 'count') {
         // take the count for element
-        mem.values.push(toObjectProperty(attr.node.value.expression.name));
-        mem.defaults = `{${attr.node.value.expression.name}, ${nodeName}, ${mem.defaults}`;
+        let exprName = attr.node.value.expression.name;
+        if (!exprName) {
+          exprName = 'count';
+        }
+        if (exprName === 'count') {
+          // if the prop expression name is also "count", copy it instead: <Plural count={count} --> <Trans count={count}
+          mem.attributesToCopy.push(attr.node);
+        } else {
+          mem.values.unshift(toObjectProperty(exprName));
+        }
+        mem.defaults = `{${exprName}, ${nodeName}, ${mem.defaults}`;
+      } else if (attr.node.name.name === 'values') {
+        // skip the values attribute, as it has already been processed into mem from existingValues
       } else if (attr.node.value.type === 'StringLiteral') {
         // take any string node as plural option
         let pluralForm = attr.node.name.name;
         if (pluralForm.indexOf('$') === 0) pluralForm = pluralForm.replace('$', '=');
         mem.defaults = `${mem.defaults} ${pluralForm} {${attr.node.value.value}}`;
       } else if (attr.node.value.type === 'JSXExpressionContainer') {
         // convert any Trans component to plural option extracting any values and components
-        const children = attr.node.value.expression.children;
+        const children = attr.node.value.expression.children || [];
         const thisTrans = processTrans(children, babel, componentStartIndex);
 
         let pluralForm = attr.node.name.name;
         if (pluralForm.indexOf('$') === 0) pluralForm = pluralForm.replace('$', '=');
 
         mem.defaults = `${mem.defaults} ${pluralForm} {${thisTrans.defaults}}`;
         mem.components = mem.components.concat(thisTrans.components);
-        mem.values = mem.values.concat(thisTrans.values);
```

Comment:
this was causing problems with duplicate keys in `values`, and I'm not sure I understand the use case and all the test cases I needed are passing, so I just removed it.
If we need to support this, then I can add code to dedupe these from `values`.